kaiax/reward/impl/getter.go

```diff
@@ -0,0 +1,395 @@
+			// Just distribute the minting reward to the proposer and stop.
+			proposer = new(big.Int).Set(minted)
+			totalFee = big.NewInt(0)
+		} else {
+			// But Simple policy had a bug where transaction fees were distributed to the proposer here at the end of block processing
+			// despite configured to non-deferred mode. To keep the backward compatibility, the buggy behavior retains until Magma.
+			proposer = new(big.Int).Add(minted, totalFee)
+		}
+		spec.Minted = new(big.Int).Set(minted)
+		spec.TotalFee = totalFee
+		spec.BurntFee = big.NewInt(0)
+		spec.Proposer = proposer
+		spec.IncReceipient(config.Rewardbase, proposer)
+		return spec, nil
+	}
+
+	// Deferred mode
+	burntFee := big.NewInt(0)
+	if config.Rules.IsMagma {
+		burntFee = getBurnAmountMagma(totalFee)
+	}
+	proposer := new(big.Int).Add(minted, totalFee)
+	proposer.Sub(proposer, burntFee)
+
+	spec.Minted = minted
+	spec.TotalFee = totalFee
+	spec.BurntFee = burntFee
+	spec.Proposer = proposer
+	spec.IncReceipient(config.Rewardbase, proposer)
+	return spec, nil
+}
+
+// getDeferredRewardFull is for non-Simple policy.
+func getDeferredRewardFull(config *reward.RewardConfig, totalFee *big.Int, si *staking.StakingInfo) (*reward.RewardSpec, error) {
+	// Non-deferred and deferred modes share most of the logic
+	// except that in non-deferred mode the block fees are considered zero.
+	var burntFee *big.Int
+	if !config.DeferredTxFee {
+		totalFee = big.NewInt(0)
+		burntFee = big.NewInt(0)
+	} else {
+		if config.Rules.IsKore {
+			burntFee = getBurnAmountKore(config, totalFee)
+		} else if config.Rules.IsMagma {
+			burntFee = getBurnAmountMagma(totalFee)
+		} else {
+			burntFee = big.NewInt(0)
+		}
+	}
+
+	// Both non-deferred and deferred modes
+	if config.Rules.IsKore {
+		return getDeferredRewardFullKore(config, totalFee, burntFee, si)
+	} else {
+		return getDeferredRewardFullLegacy(config, totalFee, burntFee, si)
+	}
+}
+
+// getDeferredRewardFullKore is for non-Simple policy and after Kore.
+func getDeferredRewardFullKore(config *reward.RewardConfig, totalFee, burntFee *big.Int, si *staking.StakingInfo) (*reward.RewardSpec, error) {
+	var (
+		spec         = reward.NewRewardSpec()
+		minted       = new(big.Int).Set(config.MintingAmount)
+		remainingFee = new(big.Int).Sub(totalFee, burntFee)
+	)
+
+	// Distribute using RewardRatio first. Unlike Legacy, fees are not distributed here
+	// because fees are exclusively allocated to proposer. Remainder goes to KIF.
+	validators, kif, kef := config.RewardRatio.Split(minted)
+	ratioRemainder := calcRemainder(minted, validators, kif, kef)
+	kif.Add(kif, ratioRemainder)
+
+	// Further distribute using Kip82Ratio. Remainder goes to proposer.
+	proposer, stakers := config.Kip82Ratio.Split(validators)
+	stakersAlloc, kip82Remainder := assignStakingRewards(config, stakers, si)
+	proposer.Add(proposer, kip82Remainder)
+	stakers.Sub(stakers, kip82Remainder)
```

Comment:
I know that this is preserved one in this modulirzation. According to [KIP-82](https://github.com/klaytn/kips/blob/main/KIPs/kip-82.md), remainder goes to the proposer, but not mentioned why stakers' allocation is decreased because no description was found for this.

Given a situation,
```proposer = 20, stakers = 20```
if the remainder is `1` by splitting process of bigint, the stakers would be `19`. Then, the remainder goes to propsoer. Thus, proposer and stakers would be `21` and `19` resp.

However, the subtraction to stakers makes a change of stakers value to `18`. Finally, both are `21` and `18`. totally `39`, which is different with initial sum of proposer and stakers. 
cc. @ian0371

Reply:
`stakers` does not change in `assignStakingRewards`, so `stakers` is still 20 before executing the line 270.
Although the sum of `stakersAlloc` would be 19.

Reply:
Ahh, I see. Then it just changes the summary, not the actual balance update. If so, the difference still seems odd to me. Can you let me know what this contributes to?

Reply:
What do you mean by "difference"? `stakers` and the sum of `stakersAlloc` are both 19 eventually, which is the exact amount to be distributed.

Reply:
I had a misconception with `stakes.Sub(stakersAlloc, kip82Remainder)`, but actually is `stakers.Sub(stakers, kip82Remainder)`. Got it. Thank you.